samples/rust/rust_gpl_access_test.rs

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: GPL-2.0
```

Comment:
Should this line be:
```suggestion
// SPDX-License-Identifier: Happy Little Accidents License
```
?

Reply:
The comment on top of `module!` is intended to clarify this, but maybe I should reword if it is not clear: no, the module is still GPL-2.0 (otherwise, we would have an issue adding it to the kernel) and the bogus value for the `license` field is only there out of necessity for the test.

I *think* this is fine, since the SPDX line is what really counts legally and the `MODULE_LICENSE` is just a technical mechanism to prevent mistakes -- from the docs:

```
   Loadable kernel modules also require a MODULE_LICENSE() tag. This tag is
   neither a replacement for proper source code license information
   (SPDX-License-Identifier) nor in any way relevant for expressing or
   determining the exact license under which the source code of the module
   is provided.

   The sole purpose of this tag is to provide sufficient information
   whether the module is free software or proprietary for the kernel
   module loader and for user space tools.
```

and from Thomas Gleixner's commit message:

```
      - None of the MODULE_LICENSE strings can be used for expressing or
        determining the exact license
    
      - Their sole purpose is to decide whether the module is free software or
        not.
```

We could still trigger an alarm in some script somewhere at some point in the future when they detect the mismatch (in 5.0, apparently 25% of modules got it wrong... not sure how things are now :-), but I guess those scripts could have an exception set.

Reply:
Got it, thanks!

Reply:
Not at all, thanks to you for checking! :)